src/pybind/mgr/tests/test_object_format.py

```diff
@@ -0,0 +1,162 @@
+        ),
+    ],
+)
+def test_to_json_many(objs: Any, json_val: str):
+    assert object_format.to_json(objs, many=True) == json_val
+
+
+@pytest.mark.parametrize(
+    "obj, yaml_val",
+    [
+        ({}, "{}\n"),
+        ({"name": "foobar"}, "name: foobar\n"),
+        ({"stuff": [1, 88, 909, 32]}, "stuff:\n- 1\n- 88\n- 909\n- 32\n"),
+        (JSONer("zebulon", "999"), "name: zebulon\nvalue: '999'\nversion: 1\n"),
+    ],
+)
+def test_to_yaml(obj: Any, yaml_val: str):
+    assert object_format.to_yaml(obj) == yaml_val
+
+
+@pytest.mark.parametrize(
+    "obj, cls, yaml_val",
+    [
+        (JSONer("yen"), JSONer, "name: yen\nvalue: {}\nversion: 2\n"),
+        ({}, JSONer, "name: ''\nvalue: {}\nversion: 2\n"),
+    ],
+)
+def test_to_yaml_cls(obj: Any, cls: Any, yaml_val: str):
+    assert object_format.to_yaml(obj, cls=cls) == yaml_val
+
+
+@pytest.mark.parametrize(
+    "objs, yaml_val",
+    [
+        ([{}], "{}\n"),
+        ([{"name": "foobar"}], "name: foobar\n"),
+        (
+            [{"name": "foobar"}, {"name": "quuxy"}],
+            ("name: foobar\n" "---\n" "name: quuxy\n"),
+        ),
+        (
+            [JSONer("zebulon", "999"), JSONer("christopher", "2891")],
+            (
+                "name: zebulon\nvalue: '999'\nversion: 1\n"
+                "---\n"
+                "name: christopher\nvalue: '2891'\nversion: 1\n"
+            ),
+        ),
+    ],
+)
+def test_to_yaml_many(objs: Any, yaml_val: str):
+    assert object_format.to_yaml(objs, many=True) == yaml_val
+
+
+@pytest.mark.parametrize(
+    "objs, cls, yaml_val",
+    [
+        ([{}], JSONer, "name: ''\nvalue: {}\nversion: 2\n"),
+        ([{"name": "foobar"}], JSONer, "name: foobar\nvalue: {}\nversion: 2\n"),
+        (
+            [{"name": "foobar"}, {"name": "quuxy"}],
+            JSONer,
+            (
+                "name: foobar\nvalue: {}\nversion: 2\n"
+                "---\n"
+                "name: quuxy\nvalue: {}\nversion: 2\n"
+            ),
+        ),
+        (
+            [JSONer("zebulon", "999"), JSONer("christopher", "2891")],
+            JSONer,
+            (
+                "name: zebulon\nvalue: '999'\nversion: 2\n"
+                "---\n"
+                "name: christopher\nvalue: '2891'\nversion: 2\n"
+            ),
+        ),
+    ],
+)
+def test_to_yaml_many_cls(objs: Any, cls: Any, yaml_val: str):
+    assert object_format.to_yaml(objs, many=True, cls=cls) == yaml_val
+
+
+def test_yaml_representer():
+    import yaml
+
+    class YAMLoid(JSONer):
+        pass
+
+        @staticmethod
+        def yaml_representer(dumper: "yaml.SafeDumper", data: Any):
+            d = dict(data.to_json())
+            d["objectType"] = "YAMLoid"
+            return dumper.represent_dict(
+                sorted(
+                    d.items(),
+                    key=lambda p: "" if p[0] == "objectType" else p[0],
```

Comment:
the `key` argument of `sorted` requires a function (callback) - https://docs.python.org/3/library/functions.html#sorted

For each item in the "list" of things to sort the `key` function will be called in order to return a more suitable type for sorting and thus determine the sort order.
We're sorting the data because it makes the tests (more) reliable.